http/transport/retry_round_tripper.go

```diff
@@ -12,8 +12,8 @@ import (
 
 // RetryRoundTripper implements a chainable round tripper for retrying requests
 type RetryRoundTripper struct {
-	RetryTransport *retry.Transport
```

Comment:
I think the issue you discovered is because of the `DefaultRetryTransport` used by more than one client. One client should not use the `DefaultRetryTransport` concurrently, can you try another fix? E.g. remove the already deprecated `DefaultRetryTransport` and only use the function `NewDefaultRetryTransport`. It would be helpful to have a test to see where the race is originating from, or at least the trace for the race.

Reply:
No, as stated above the race I discovered is specifically only related to a single client using the bricks `transport.NewDefaultTransportChain()` when two go routines do requests through this client concurrently. For better understanding I added a test before applying my fix (aa41025). If you do a `make test` on this commit, it will fail due to a race condition.